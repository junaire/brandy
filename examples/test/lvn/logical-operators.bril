# ARGS: -f

@main(arg1: bool, arg2: bool) {
  t: bool = const true;
  f: bool = const false;

  constant_fold1: bool = and f t;
  constant_fold2: bool = and t f;
  constant_fold3: bool = or t f;
  constant_fold4: bool = or f t;
  constant_fold5: bool = not t;
  constant_fold6: bool = not f;

  should_fold1: bool = and f arg1;
  should_fold2: bool = and arg1 f;
  should_fold3: bool = or t arg1;
  should_fold4: bool = or arg1 t;

  no_fold1: bool = and t arg1;
  no_fold2: bool = and arg1 t;
  no_fold3: bool = or f arg1;
  no_fold4: bool = or arg1 f;
  no_fold5: bool = and arg1 arg2;
  no_fold6: bool = or arg1 arg2;
  no_fold7: bool = not arg1;
}
