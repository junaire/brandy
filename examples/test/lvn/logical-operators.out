@main(arg1: bool, arg2: bool) {
  t: bool = const true;
  f: bool = const false;
  constant_fold1: bool = const false;
  constant_fold2: bool = const false;
  constant_fold3: bool = const true;
  constant_fold4: bool = const true;
  constant_fold5: bool = const false;
  constant_fold6: bool = const true;
  should_fold1: bool = const false;
  should_fold2: bool = const false;
  should_fold3: bool = const true;
  should_fold4: bool = const true;
  no_fold1: bool = and t arg1;
  no_fold2: bool = and arg1 t;
  no_fold3: bool = or f arg1;
  no_fold4: bool = or arg1 f;
  no_fold5: bool = and arg1 arg2;
  no_fold6: bool = or arg1 arg2;
  no_fold7: bool = not arg1;
}
